src/main/java/dev/galacticraft/impl/internal/mixin/gear/TamableAnimalMixin.java

```diff
@@ -0,0 +1,100 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package dev.galacticraft.impl.internal.mixin.gear;
+
+import dev.galacticraft.api.accessor.GearInventoryProvider;
+import dev.galacticraft.impl.internal.inventory.MappedInventory;
+import dev.galacticraft.mod.Constant;
+import dev.galacticraft.mod.Galacticraft;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.Tag;
+import net.minecraft.world.Container;
+import net.minecraft.world.SimpleContainer;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.OwnableEntity;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.TamableAnimal;
+import net.minecraft.world.entity.animal.Cat;
+import net.minecraft.world.entity.animal.Parrot;
+import net.minecraft.world.entity.animal.Wolf;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+
+@Mixin(TamableAnimal.class)
+public abstract class TamableAnimalMixin extends Entity implements GearInventoryProvider, OwnableEntity {
+
+    private final @Unique SimpleContainer gearInv = this.galacticraft_createGearInventory();
+    private final @Unique Container tankInv = MappedInventory.create(this.gearInv, 2);
+    private final @Unique Container thermalArmorInv = MappedInventory.create(this.gearInv, 3);
+    private final @Unique Container accessoryInv = MappedInventory.create(this.gearInv, 0, 1);
+
+    TamableAnimalMixin() {
+        super(null, null);
+    }
+
+    @Override
+    public SimpleContainer galacticraft$getGearInv() {
+        return this.gearInv;
+    }
+
+    @Override
+    public Container galacticraft$getOxygenTanks() {
+        return this.tankInv;
+    }
+
+    @Override
+    public Container galacticraft$getThermalArmor() {
+        return this.thermalArmorInv;
+    }
+
+    @Override
+    public Container galacticraft$getAccessories() {
+        return this.accessoryInv;
+    }
+
+    @Override
+    public long galacticraft$oxygenConsumptionRate() {
+        TamableAnimal animal = (TamableAnimal) (Object) this;
+        LivingEntity owner = animal.getOwner();
+        if (owner != null && !owner.isSpectator() && animal.distanceToSqr(owner) >= 4096.0) {
```

Comment:
I assume we wouldn't want animals to die while players are disconnected?
```suggestion
        if (owner == null || (!owner.isSpectator() && animal.distanceToSqr(owner) >= 4096.0)) {
```

Reply:
Definitely not, thanks!